cmd/agent/workspace/up.go

```diff
@@ -420,11 +529,33 @@ func installDocker(log log.Logger) error {
 		shellCommand := exec.Command("sh", "-c", scripts.InstallDocker)
 		shellCommand.Stdout = writer
 		shellCommand.Stderr = writer
-		err := shellCommand.Run()
-		if err != nil {
+		err = shellCommand.Run()
+	}
+	return err
+}
+
+func configureDockerDaemon(ctx context.Context, log log.Logger) (err error) {
+	log.Info("Configuring docker daemon ...")
+	// Enable image snapshotter in the dameon
+	var daemonConfig = []byte(`{
+		"features": {
+			"containerd-snapshotter": true
+		}
+	}`)
+	// Check rootless docker
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	if _, err = os.Stat(fmt.Sprintf("%s/.config/docker", homeDir)); !errors.Is(err, os.ErrNotExist) {
+		err = os.WriteFile(fmt.Sprintf("%s/.config/docker/daemon.json", homeDir), daemonConfig, 0644)
+	}
+	// otherwise assume default
+	if err != nil {
+		if err = os.WriteFile("/etc/docker/daemon.json", daemonConfig, 0644); err != nil {
 			return err
 		}
 	}
-
-	return nil
+	// reload docker daemon
+	return exec.CommandContext(ctx, "pkill", "-HUP", "dockerd").Run()
```

Comment:
is this the best way to reload the docker daemon? I was surprised docker did not have a command for this and I didn't think I could assume `systemd` was being used to manage docker

Reply:
this looks scary 😬

Reply:
can't think of a better way to do it though, unless we'd start this build container with already mounted config? is that even possible?

Reply:
I agree, maybe I could try to detect how dockerd is managed (systemd,systemctl etc) using `uname` and if not available use this as a last resort? Yes that is possible, I thought of the mounted config but this would be provider specific and need to be implemented by each machine provider, gcloud, aws, digital ocean etc. :/ At least with HUP it's a reload and not restart but it's a bit "under the hood"